src/server_manager/web_app/digitalocean_account.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright 2021 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {DigitalOceanSession, DropletInfo} from "../cloud/digitalocean_api";
+import {DigitalOceanServer, GetCityId} from "./digitalocean_server";
+import * as server from "../model/server";
+import * as crypto from "../infrastructure/crypto";
+import * as digitalocean from "../model/digitalocean";
+import * as do_install_script from "../install_scripts/do_install_script";
+
+// Tag used to mark Shadowbox Droplets.
+const SHADOWBOX_TAG = 'shadowbox';
+const MACHINE_SIZE = 's-1vcpu-1gb';
+
+export interface ShadowboxSettings {
+  imageId: string;
+  metricsUrl: string;
+  sentryApiUrl?: string;
+  watchtowerRefreshSeconds?: number;
+}
+
+export class DigitalOceanAccount implements digitalocean.DigitalOceanAccount {
+  private servers: DigitalOceanServer[] = [];
+
+  constructor(
+      private digitalOcean: DigitalOceanSession, private shadowboxSettings: ShadowboxSettings,
+      private debugMode: boolean) {}
+
+  async getName(): Promise<string> {
+    return (await this.digitalOcean.getAccount())?.email;
+  }
+
+  async getStatus(): Promise<digitalocean.Status> {
+    const account = await this.digitalOcean.getAccount();
+    if (account.status === 'active') {
+      return digitalocean.Status.ACTIVE;
+    }
+    if (!account.email_verified) {
+      return digitalocean.Status.EMAIL_UNVERIFIED;
+    }
+    return digitalocean.Status.INVALID_BILLING_INFORMATION;
```

Comment:
Maybe MISSING_BILLING_INFORMATION is more accurate?

Reply:
Done